lib/api/apiUtils/object/abortMultipartUpload.js

```diff
@@ -98,6 +126,13 @@ function abortMultipartUpload(authInfo, bucketName, objectKey, uploadId, log,
             }
             // flatten the array
             locations = [].concat(...locations);
+
+            if (objectMD?.location) {
+                const objectLocationLeft = objectMD.location.filter(loc =>
```

Comment:
nit: "left" looks like the directoin, maybe best to rename to something more intuitive, maybe `remainingObjectLocations`

more importantly, this can be quite CPU intensive for large objects (`n^2` complexity, with `n` up to 10000): need to make this more efficient...

Reply:
Maybe a more optimized approach for this would be to use a map ?

```
if (objectMD?.location) {
    const locationMap = new Map(locations.map(loc => [loc.key, true]));
    const objectLocationLeft = objectMD.location.filter(loc => !locationMap.has(loc.key));
    locations = locations.concat(objectLocationLeft);
}
```

Reply:
since we are building a list of locations, maybe we could directly use a `map` instead (and pass it to async.eachLimit directly, to process each "pair"?)

Reply:
Pushed a more optimized version here
https://github.com/scality/cloudserver/pull/5691/commits/e9ae9c22ec15e3ae2965eef3c3d1115d15a69c2a

FOr locations being a Set directly, I can do it, but as we'll need to map over it anyway for getting keys, we'll end up needing an array (as `map` function does not exist for Sets)